### PR TITLE
Fix teams on random maps

### LINKED ISSUE #1233: Player doesn't participate in game
1) Open random map menu
3) Keep all map settings random
4) Open additional settings
5) Select pink flag (the last one) for human player
6) Start game

Expected: game is started - player has pink flag
Actual: game is started between AI players only - they are playing in background

Note: if RMG will generate map for 8 players issue will not be reproduced, so it's better to generate M+U or L map

### LINKED ISSUE #3044: RMG settings bug. Tan player banished on map start
On 2SM2i(2), with 3 human or CPU and 0 cpu, at game start TAN loses.....

**Describe the bug**

image

### PR DESCRIPTION
**Pull request by DjWarmonger**
- [x] Do not allow 0 or 1 teams
- [x] Do not allow player to set only 1 team in GUI
- [x] Fix player and team settings not being propagated to RMG from GUI
- [x] Fix players removed at game start 
- [x] Remove misleading error about "no info for player X found"
- [ ] Fix #1233 
- [x] Fix #3044 

Now, it looks like original logic and text description of the number of players is messed up and not consistent with template parameters. Original rmg.txt contains the following info:

- Minimum, maximum human positions (1 to N)
- Minimum, maximum total positions (2 to N)

- [x] Change template parameters logic only
- [x] Infer GUI values indirectly from these fixed parameters

### CODE REVIEW

File: lib/rmg/CMapGenOptions.h [hunk 1]
```diff
@@ -53,6 +53,9 @@ class DLL_LINKAGE CMapGenOptions
 		TeamID getTeam() const;
 		void setTeam(const TeamID & value);
 
+		/// Constant for a random town selection.
+		static const si32 RANDOM_TOWN = -1;
```
Comment (IvanSavenko) [inline lib/rmg/CMapGenOptions.h:57]: There is already FactionID::RANDOM for such cases

**Comment by DjWarmonger** [comment]
Okay, template format is fixed and all options work as intended.

I'll just leave #1233 for now, and it feels non-trivial and it's a very minor feature as the same time. Not worth delaying the PR.

**Comment by IvanSavenko** [comment]
@DjWarmonger Merge?

**Comment by DjWarmonger** [comment]
@IvanSavenko Conflicts :v

**Comment by IvanSavenko** [comment]
If that's the only problem I can merge it tomorrow

**Comment by DjWarmonger** [comment]
Let me fix that

**Pull request merged**